Introduce option to display shorter author data on EDS item detail page

alexklbuckley opened the pull request:
- Add DetailPageFormat config to EDS.ini
- Add short author display to EDS core.phtml files
- Follow-up to PR3841

Sponsored-By: Auckland University of Technology, New Zealand

alexklbuckley committed:
Introduce option to display shorter author data on EDS item detail page

- Add DetailPageFormat config to EDS.ini
- Add short author display to EDS core.phtml files

Sponsored-By: Auckland University of Technology, New Zealand

demiankatz commented on themes/bootstrap3/templates/RecordDriver/EDS/core.phtml:
```diff
@@ -77,10 +77,33 @@
-          <th><?=empty($item['Label']) ? '' : $this->transEsc($item['Label']) . ':'?></th>
-          <td class="record__biblio-value"><?=$this->driver->linkUrls($item['Data'])?></td>
-        </tr>
+          <?php if ('Au' == $item['Group'] && strtolower($edsConfig->AuthorDisplay->ResultListFormat ?? 'Long') == 'short'): ?>
```
I don't think $edsConfig is defined in the context of this template, unless I am overlooking something.

Also, the .ini file says the default is Short, but the default here is Long. We should pick one and be consistent.

alexklbuckley replied:
thanks @demiankatz ! sorry about the delay.

I've pushed a follow-up addressing both of your points here.

demiankatz reviewed (changes requested):
Thanks, @alexklbuckley! I haven't tried hands-on testing yet because I suspect it's not quite working yet based on my comments below. I'll take a closer look once you have addressed this review.

alexklbuckley committed:
Merge branch 'vufind-org:dev' into eds_detail_short_author_data

alexklbuckley committed:
Merge branch 'vufind-org:dev' into eds_detail_short_author_data

alexklbuckley committed:
Introduce option to display shorter author data on EDS item detail page: Implement review requests

- Add fetching EDS config to EDS/core.phtml
- Change documentation in EDS.ini to note that the DetailPageFormat config defaults to Long.

Sponsored-By: Auckland University of Technology, New Zealand

berndoberknapp commented:
I would suggest to not show the AuInfo item when DetailPageFormat = 'Short' is used. When the AuInfo item is present it contains author affiliation information referenced from the Au item.

demiankatz reviewed (changes requested):
Thanks, @alexklbuckley! I'm marking this back to "request changes" until @berndoberknapp's suggestion is addressed, and then I'll take a closer look myself. :-)

alexklbuckley committed:
Merge branch 'vufind-org:dev' into eds_detail_short_author_data

alexklbuckley committed:
Introduce option to display shorter author data on EDS item detail page

- Add default short author name delimiters to EDS.css.
- Ensure only
  'BibRecord/BibRelationships/HasContributorRelationships/*PersonEntity/Name/NameFull'
data is shown when DetailPageFormat='Short'.
- Ensure the current author data (containing author affiliations) is
  displayed when DetailPageFormat='Long'.

Sponsored-By: Auckland University of Technology, New Zealand

alexklbuckley commented:
Hi @demiankatz and @berndoberknapp , I have been doing some local testing of this enhancement and have pushed a follow-up fixing a few bugs that I noticed.

Now when DisplayPageFormat is set to:
- 'Short' - Then 'BibRecord/BibRelationships/HasContributorRelationships/*PersonEntity/Name/NameFull' data is displayed. Just like  for https://github.com/vufind-org/vufind/pull/3841
- 'Long' - Then the current author data that is displayed on upstream dev branch is displayed.

Can you please confirm for me Bernd if this is the functionality fix you were wanting in https://github.com/vufind-org/vufind/pull/3881#issuecomment-2325271586 ?

demiankatz committed:
Merge branch 'dev' into eds_detail_short_author_data

demiankatz committed:
Less redundant config comments.

demiankatz committed:
Simplify/synchronise CSS.

demiankatz committed:
Whitespace tweaks.

demiankatz reviewed (approved):
@alexklbuckley, I pushed up a few minor adjustments:

1.) I combined some of the comments in EDS.ini to reduce redundancy
2.) I combined some of the CSS in EDS.css to reduce redundancy
3.) I added missing EDS.css changes to the bootstrap5 theme
4.) I tweaked some whitespace in the core.phtml template

I'm approving this because it seems to work correctly for me, but I'll wait for @berndoberknapp's approval before actually merging in case there are any further details I have overlooked.

alexklbuckley commented:
> @alexklbuckley, I pushed up a few minor adjustments:
> 
> 1.) I combined some of the comments in EDS.ini to reduce redundancy 2.) I combined some of the CSS in EDS.css to reduce redundancy 3.) I added missing EDS.css changes to the bootstrap5 theme 4.) I tweaked some whitespace in the core.phtml template
> 
> I'm approving this because it seems to work correctly for me, but I'll wait for @berndoberknapp's approval before actually merging in case there are any further details I have overlooked.

Thanks @demiankatz !

berndoberknapp commented:
I won't be able to test the changes before next week, but my comment regaring AuInfo doesn't seem to be addressed yet?

demiankatz commented:
> I won't be able to test the changes before next week, but my comment regaring AuInfo doesn't seem to be addressed yet?

I'm not very familiar with the API responses. Is AuInfo a different `$item['group']` value that we might encounter? Is the idea that if we're set to short mode, this information will be irrelevant and should be bypassed? If that's the case, could you perhaps share a couple of example record IDs that include this element?

berndoberknapp commented:
> I'm not very familiar with the API responses. Is AuInfo a different `$item['group']` value that we might encounter? Is the idea that if we're set to short mode, this information will be irrelevant and should be bypassed? If that's the case, could you perhaps share a couple of example record IDs that include this element?

Yes, AuInfo is a separate group with additional information about the authors. While looking for examples I noticed that there are different cases. For many EBSCOhost databases like Academic Search or Business Source Au includes references to AuInfo which only contains "Author Affiliations", and in that case AuInfo doesn't make much sense when Short is used (e.g. aph,178692351). But there are databases like Psyndex where AuInfo contains information like "Author ID", "Author Email" or "Country of First Author" and where the author name is repeated, so in that case it would make sense to display the information (e.g. pdx,0421957). I can try to check whether this is consistent across databases, but not before next week.

demiankatz commented:
Thanks, @berndoberknapp! It sounds like it could get complicated to try to display AuInfo only for certain databases. It may be sufficient to just hide it in "short" mode as you propose -- we would just likely want to make note of this in the config settings.

demiankatz committed:
Clarify defaults.

demiankatz committed:
Hide AuInfo in short mode.

demiankatz commented:
I've just pushed up a change to hide the AuInfo group in short author display mode. I think this is probably okay since most people will use the defaults (short in search results, long on record view) and it makes the most sense in that context.... but of course I'm open to change if people have other ideas. @alexklbuckley and @berndoberknapp, let me know if you're comfortable merging things in this current form!

alexklbuckley commented:
> I've just pushed up a change to hide the AuInfo group in short author display mode. I think this is probably okay since most people will use the defaults (short in search results, long on record view) and it makes the most sense in that context.... but of course I'm open to change if people have other ideas. @alexklbuckley and @berndoberknapp, let me know if you're comfortable merging things in this current form!

Thank you very much @demiankatz !

Yes, from my perspective, I am happy for this to be merged in it's current form :)

berndoberknapp commented:
> I've just pushed up a change to hide the AuInfo group in short author display mode. I think this is probably okay since most people will use the defaults (short in search results, long on record view) and it makes the most sense in that context.... but of course I'm open to change if people have other ideas. @alexklbuckley and @berndoberknapp, let me know if you're comfortable merging things in this current form!

That seems to be the most sensible solution, I'm of course fine with merging it.

demiankatz commented:
Thanks, @alexklbuckley and @berndoberknapp -- merging now!

alexklbuckley commented:
Thanks @demiankatz and @berndoberknapp !
